### PR TITLE
feat(identity-providers): JIT `Proxy Mode` passthrough validation

### PR DESCRIPTION
**Pull request by yasinmiran**
### Related Issues
- Issue `#1` or (None)

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there's any)
- [ ] Unit tests provided. (Add links if there's any)
- [ ] Integration tests provided. (Add links if there's any)

### Related PRs
- Related PR `#1` or (None)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/applications/components/settings/sign-on-methods/step-based-flow/authenticators.tsx [hunk 1]
```diff
@@ -229,6 +255,30 @@ export const Authenticators: FunctionComponent<AuthenticatorsPropsInterface> = (
...
+                    .filter((auth: GenericAuthenticatorInterface & { provisioning: ProvisioningInterface }) => {
+                        return !auth?.provisioning?.jit?.isEnabled
+                    })
+                    .map(({ name }) => name)
+                    .filter(Boolean)
+            );
+            const cleanedFilters = selectedAuthenticators.filter(({ name }) => !conflictingHandlers.has(name));
+            onAuthenticatorSelect([ ...cleanedFilters, selectedAuthenticator ]);
+            setSelectedAuthenticators([ ...cleanedFilters, selectedAuthenticator ]);
+            return;
```
Comment (thivi) [inline apps/console/src/features/applications/components/settings/sign-on-methods/step-based-flow/authenticators.tsx:279]:
```suggestion

            return;
```

File: apps/console/src/features/identity-providers/api/identity-provider.ts [hunk 2]
```diff
@@ -522,8 +565,43 @@ export const updateJITProvisioningConfigs = (
                 error.code,
                 error.request,
                 error.response,
-                error.config);
+                error.config
+            );
+        });
+};
+
+
```
Comment (thivi) [inline apps/console/src/features/identity-providers/api/identity-provider.ts:573]:
```suggestion
```

File: apps/console/src/features/identity-providers/components/forms/jit-provisioning-configuration-form.tsx [hunk 3]
```diff
@@ -112,113 +132,244 @@ export const JITProvisioningConfigurationsForm: FunctionComponent<JITProvisionin
...
+    useEffect(() => {
+
+        setFetchingConnectedApps(true);
+
+        getIDPConnectedApps(idpId)
+            .then(async (response: ConnectedAppsInterface) => {
+                if (response.count === 0) {
+                    setCannotModifyProxyModeDueToConnectApps(false);
+                    setFetchingConnectedApps(false);
+                    return;
```
Comment (thivi) [inline apps/console/src/features/identity-providers/components/forms/jit-provisioning-configuration-form.tsx:147]:
```suggestion

                    return;
```

**Review by thivi — APPROVED** [review]

File: apps/console/src/features/identity-providers/components/forms/jit-provisioning-configuration-form.tsx (hunk 3 above)
Comment (yasinmiran) [reply apps/console/src/features/identity-providers/components/forms/jit-provisioning-configuration-form.tsx:148]: Addressed e3f53e5

File: apps/console/src/features/identity-providers/api/identity-provider.ts (hunk 2 above)
Comment (yasinmiran) [reply apps/console/src/features/identity-providers/api/identity-provider.ts]: Addressed with e3f53e5

File: apps/console/src/features/applications/components/settings/sign-on-methods/step-based-flow/authenticators.tsx (hunk 1 above)
Comment (yasinmiran) [reply apps/console/src/features/applications/components/settings/sign-on-methods/step-based-flow/authenticators.tsx:280]: Addressed with e3f53e5

**Pull request merged**